drivers/soundwire/intel_ace2x.c

```diff
@@ -148,10 +162,39 @@ static int intel_ace2x_bpt_open_stream(struct sdw_intel *sdw, struct sdw_slave *
 	sdw->bpt_ctx.pdi1_buffer_size = pdi1_buffer_size;
 	sdw->bpt_ctx.num_frames = num_frames;
 	sdw->bpt_ctx.data_per_frame = data_per_frame;
-	tx_dma_bandwidth = div_u64((u64)pdi0_buffer_size * 8 * (u64)prop->default_frame_rate,
-				   num_frames);
-	rx_dma_bandwidth = div_u64((u64)pdi1_buffer_size * 8 * (u64)prop->default_frame_rate,
-				   num_frames);
+
+	alignment = hda_sdw_bpt_get_buf_size_alignment(rx_dma_bandwidth);
+
+	if (command) { /* read */
+		/*
+		 * Add at most 3 fake frames for read command to make the pdi0_buffer_size a
+		 * mutiple of rx alignment and pdi1_buffer_size a mutiple of TX_BUF_ALIGNMENT.
+		 */
+		for (i = 1; i < data_per_frame * 3; i++) {
```

Comment:
this looks suspect, looks like you are mixing frame counter and data counter. You may want to use a better index name than 'i'...

Reply:
`i` is the fake data size and I need to set a maximum number which is currently 3 frames. But I am not quite sure if at most 3 fake frames is reasonable or not.

Reply:
looks like you'd need something like a LCM to find out how many invalid/inactive BRA/BTP frames to add. An iterative loop could fail to find the right amount, no?

Reply:
Yeah, it would be better if I calculate the required frames instead of finding it with a loop.